### PR TITLE
[NativeAOT-LLVM] Update user-facing documentation

### PR DESCRIPTION
**Pull request by SingleAccretion**
This is a fairly comprehensive revamp of our user-facing documentation to be more focused on the NativeAOT-LLVM use cases. We inherited much of the documentation from the NativeAOT branch, but not all of it is relevant (and some of it is duplicative).

I also made the `samples` build and run with NativeAOT-LLVM. In particular, the native library sample now houses a little calculator showing how to interact with the Emscripten-generated JS code.

I have also included two small changes to functional code:
1) Make the build not produce an `.html` file when building a library.
2) Make use of a slightly different mechanism for invoking `clang` that looks better with the new MSBuild logger:

image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,25 @@
...
+
 ## How to Engage, Contribute and Provide Feedback
-Some of the best ways to contribute are to try things out, file bugs, and join in design conversations.
 
-Looking for something to work on? The [_help wanted_](https://github.com/dotnet/runtimelab/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22++label%3Aarea-NativeAOT+) issues are a great place to start.
+Some of the best ways to contribute are to try things out, file bugs, and join in design conversations.
 
-[![Join the chat at https://gitter.im/dotnet/corert](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/dotnet/corert?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
+Looking for something to work on? The [_help wanted_](https://github.com/dotnet/runtimelab/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22++label%3Aarea-NativeAOT-LLVM+) issues are a great place to start.
```
Comment (SingleAccretion) [inline README.md:20]: There is only one such issue right now though, and it's #1828...

File: samples/NativeLibrary/NativeLibraryHost.html [hunk 2]
```diff
@@ -0,0 +1,90 @@
+<!doctype html>
```
Comment (SingleAccretion) [inline samples/NativeLibrary/NativeLibraryHost.html:1]: Here's how this looks:

image

**Comment by SingleAccretion** [comment]
@dotnet/nativeaot-llvm

**Review by yowl — APPROVED** [review]
All good, thanks!

**Review by jkotas — APPROVED** [review]
Thank you!

**Pull request merged**